daria.khokhryakova/task-3/internal/parser/json.go

```diff
@@ -0,0 +1,55 @@
+package parser
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/DariaKhokhryakova/task-3/internal/models"
+)
+
+func ProcessCurrencies(valCurs *models.ValCurs) ([]models.CurrencyResult, error) {
+	results := valCurs.Currencies
+
+	sort.Slice(results, func(i, j int) bool {
+		return results[i].Value > results[j].Value
+	})
+
+	return results, nil
+}
+
+func panicErr(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func SaveJSONResults[T any](results []T, outputPath string, dirPerm os.FileMode) error {
+	dir := filepath.Dir(outputPath)
+
+	err := os.MkdirAll(dir, dirPerm)
+	if err != nil {
+		return fmt.Errorf("create dir: %w", err)
+	}
+
+	file, err := os.Create(outputPath)
+	if err != nil {
+		return fmt.Errorf("create file: %w", err)
+	}
+
+	defer func() {
```

Comment:
```suggestion
	defer panicIfErr(file.Close())
```

просто поаккуратнее и покороче

Reply:
но в целом это не так принципиально